### PR TITLE
FIX: Set active input handler to InputManager when package is removed (ISXB-675)

### PR DESCRIPTION
**Pull request by ritamerkl**
### Description

This change aims to fix [this](https://jira.unity3d.com/browse/ISXB-675) bug ticket, where the active input handling options can be set independently of the presence of the package.
Now the input handling is set to InputManager once the package is removed. The second part of this fix lands on trunk, where setting the input handling to "InputSystem" or "Both" can only be done when the package is installed.

The fix is available vor Unity version 2020.2 and above, for the sake of the PackageManager package register event being only available from 2020.2 and above.

### Testing status & QA

Manual testing.

### Overall Product Risks

- Complexity: low
- Halo Effect: low risk

### Comments to reviewers

_Please describe any additional information such as what to focus on, or historical info for the reviewers._

### Checklist

Before review:

- [ x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Editor/InputSystemPackageControl.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using System.Collections.ObjectModel;
+using UnityEditor;
+using UnityEditor.PackageManager;
+
+
+namespace UnityEngine.InputSystem.Editor
+{
+    /// <summary>
+    /// Force restart if InputSystem package is removed to activate and initialize it on managed side.
+    /// Set Project Settings input handling to InputManager once the package is removed.
```
Comment (jfreire-unity) [inline Packages/com.unity.inputsystem/InputSystem/Editor/InputSystemPackageControl.cs]:
```suggestion
    /// Forces the Editor to restart if the InputSystem package is removed to activate and initialize it on the managed side.
    /// Automatically sets "Project Settings > Player > Active Input Handling" to "Input Manager" once the package is removed.
```

File: Packages/com.unity.inputsystem/InputSystem/Editor/InputSystemPackageControl.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    /// </summary>
+    internal class InputSystemPackageControl
+    {
+        const string packageName = "com.unity.inputsystem";
+
+        [InitializeOnLoadMethod]
+        static void  SubscribePackageManagerEvent()
+        {
+            //there's a number of cases where it might not be called, for instance if the user changed the project manifest and deleted the Library folder before opening the project
```
Comment (jfreire-unity) [inline Packages/com.unity.inputsystem/InputSystem/Editor/InputSystemPackageControl.cs]: Nitpick: Start the comment with capital letter

File: Packages/com.unity.inputsystem/InputSystem/Editor/InputSystemPackageControl.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            {
+                if (package.name == packageName)
+                    return true;
+            }
+            return false;
+        }
+
+        private static void HandleInputSystemRemoved()
+        {
+            //set input handling to InputManager
```
Comment (jfreire-unity) [inline Packages/com.unity.inputsystem/InputSystem/Editor/InputSystemPackageControl.cs]: Nitpick: Start the comment with capital letter

File: Packages/com.unity.inputsystem/InputSystem/Editor/InputSystemPackageControl.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            EditorPlayerSettingHelpers.newSystemBackendsEnabled = false;
+            if (EditorUtility.DisplayDialog("Unity editor restart required", "You've removed the input system package. This requires a restart of the Editor.", "Restart Editor", "Ignore (Not recommended)"))
```
Comment (jfreire-unity) [inline Packages/com.unity.inputsystem/InputSystem/Editor/InputSystemPackageControl.cs]:
```suggestion
            if (EditorUtility.DisplayDialog("The Unity Editor needs to be restarted", "You've removed the Input System package. This requires a restart of the Unity Editor.", "Restart the Editor", "Ignore (Not recommended)"))
```

**Review by jfreire-unity — APPROVED** [review]
Nice one, thanks for dealing with this. Code looks fine to me.
I only made some nitpick suggestions in some comments text and the dialog text. Happy to approve if QA goes well.

**Comment by Pauliusd01** [comment]
After latest commit getting this 
image

**Review by Pauliusd01 — APPROVED** [review]
Approving only if the last commit is fixed. Here's what I checked:

Removing the IS package with each of the 3 backend settings.
Updating the package instead of removing it (it did not mess with the backend as expected)
Removing the package through the manifest file while the Editor is open (the backend was changed correctly)
Removing the package through the manifest file while the Editor is closed (the backend did not change but that seems to be known)

**Pull request merged**